File: scripts/load_texts.R
library(dplyr)

read_text_files <- function(){
  listAtas <- list.files(path = "../atas", pattern = ".txt", all.files = TRUE, full.names = TRUE)
  listText <- c()
  for (ata in listAtas) {
    lines <- readLines(con = ata, encoding = "UTF-8")
    lines <- paste(lines, collapse = "\n")
    listText <- c(listText, lines)
  }
  return(listText)
}

return_data_frame <- function(){
  listText <- read_text_files()
  df <- read.csv2("../decisions.csv", sep = ",")
  df[df$meeting == 45 & df$decision == "keep", "meeting"] <- NA
  df <- df %>% na.omit()
  df <- df %>% arrange(meeting)
  
  df$text <- listText
  colnames(df) <- c("meeting", "selic", "decision", "text")
  
  df2 <- read.csv2("../copom_dates.csv", sep = ",", stringsAsFactors = FALSE)
  colnames(df2) <- c("meeting", "begin_date", "end_date", "publish_date")
  
  df <- df %>% left_join(df2, by = "meeting")
  rm(df2)
  df$begin_date <- as.Date(df$begin_date)
  df$end_date <- as.Date(df$end_date)
  df$publish_date <- as.Date(df$publish_date)
  df$selic <- as.numeric(df$selic)
  return(df)
}

Mystopwords <- c("ainda", "ante", "após", "desde", "dessa", "dessas", "desse", "desses", "disso",
                 "dez", "doze", "é", "enquanto", "p", "r", "respectivamente", "relação", "sobre", "isa",
                 "janeiro", "fevereiro", "março", "abril", "maio", "junho", 
                 "julho", "agosto", "setembro", "outubro", "novembro", "dezembro", 
                 "mês", "meses", "ano", "anos", as.character(0:9), tm::stopwords("pt"))

File: scripts/load_texts.py
import os
import re
import nltk
import pandas as pd
import numpy as np

Mystopwords = ['ainda', 'ante', 'desde', 'enquanto', 'p', 'r', 'sobre',
'janeiro', 'fevereiro', 'março', 'abril', 'maio', 'junho', 'julho', 'agosto', 'setembro',  'outubro', 'novembro', 'dezembro', 'mês', 'meses', 'ano', 'anos'] + [str(i) for i in range(10)] + nltk.corpus.stopwords.words('portuguese')


def read_text_files():
    listAtas = os.listdir("../atas")
    corpus = []
    meeting = []
    for ata in listAtas:
        meeting.append(int(re.search("[0-9]+", ata).group()))
        with open("../atas/" + ata, 'rt', encoding='utf-8') as f:
            lines = f.readlines()
            if lines:
                lines = ' '.join(lines)
                corpus.append(lines)
    return [corpus, meeting]


def return_data_frame():
    corpus, meeting = read_text_files()
    dfCorpus = pd.DataFrame(corpus, index=meeting, columns=["text"])
    del corpus, meeting
    decisions = pd.read_csv("../decisions.csv")
    decisions = decisions.dropna()
    dfAux = decisions[decisions.meeting == 45]
    decisions = decisions.drop(dfAux.index[1])
    decisions.index = np.int64(decisions.meeting.values)
    dfCorpus = dfCorpus.join(decisions)
    dates = pd.read_csv("../copom_dates.csv")
    dates.index = dates.minute
    dfCorpus = dfCorpus.join(dates)
    dfCorpus = dfCorpus.sort_index()
    return dfCorpus
